platform/pensando/sonic-platform-modules-dpu/sonic_platform/helper.py

```diff
@@ -0,0 +1,175 @@
+                docker_image_name = file.read()
+            return docker_image_name.replace('\n', '')
+        except:
+            return "dpu:v1"
+
+    def get_dpu_docker_imageID(self):
+
+        docker_image_name = self.get_dpu_docker_image_name()
+        cmd = f'docker ps | grep {docker_image_name}'
+        try:
+            output = self.runCMD(cmd)
+            pattern = f"[a-f0-9]*[\s]*{docker_image_name}"
+            docker_image_id = re.findall(pattern, output)[0].split(" ")[0]
+            return docker_image_id
+        except:
+            return None
+
+    def run_docker_cmd(self, inp_cmd):
+        docker_image_id = self.get_dpu_docker_imageID()
+        if docker_image_id == None:
+            return None
+        cmd = f"docker exec {docker_image_id} "
+        cmd += inp_cmd
+        return self.runCMD(cmd)
+
+    def runCMD(self,cmd):
+        try:
+            ret = subprocess.getoutput(cmd)
+        except Exception as e:
+            print("Exception in runCMD due to %s, cleaning up and exiting\n" % str(e))
+            sys.exit(2)
+        return ret
+
+    def get_board_id(self):
+        try:
+            board_id_hex = self.runCMD("cpldapp -r 0x80")
+            if board_id_hex:
+                board_id = int(board_id_hex, 16)
+                return board_id
+        except Exception as e:
+            print(f"Failed to get board id due to {e}")
+            return 0
+
+    def readline_txt_file(self, path):
+        try:
+            with open(path, 'r') as f:
+                output = f.readline()
+            return output.strip('\n')
+        except Exception:
+            pass
+        return ''
+
+    def read_eeprom_qsfp(self, i2c_num, reg):
+        cmd = f"i2cdump -y {i2c_num} {reg} b"
+        output = self.runCMD(cmd)
+        eeprom_dump = [line.split(" ")[1:17] for line in output.split("\n")[1:]]
+        eeprom_raw = []
+        for line in eeprom_dump:
+            eeprom_raw.extend([int(byte,16) for byte in line])
+        eeprom_byte_array = bytearray(eeprom_raw)
+        return eeprom_byte_array
+
+    def read_eeprom_qsfp_file(self, i2c_num, reg):
+        eeprom_file = f"/usr/share/sonic/platform/qsfp-eeprom-{i2c_num}.bin"
+        if self.is_host():
+            platform = self.get_platform()
+            eeprom_file = f"/usr/share/sonic/device/{platform}/qsfp-eeprom-{i2c_num}.bin"
+        try:
+            with open(eeprom_file, "rb") as file:
+                read_data = bytearray(file.read())
+                return read_data
+        except:
+            return self.read_eeprom_qsfp(i2c_num, reg)
+
+    def setup_cpldapp(self):
+        cmd = "cp /usr/share/sonic/platform/cpldapp /usr/local/bin"
+        self.runCMD(cmd)
+        cmd = "cp /usr/share/sonic/platform/libpal.so /lib/libpal.so"
+        self.runCMD(cmd)
+        cmd = "cp /usr/share/sonic/platform/liblogger.so /lib/liblogger.so"
+        self.runCMD(cmd)
+
+    def check_xcvrs_present(self):
```

Comment:
@shanshri please use sonic design where Xcvrd is managing the optics and detecting the optics presence

Reply:
Removing xcvrd related code as mentioned above